api/v1alpha3/cluster_types.go

```diff
@@ -190,7 +191,7 @@ func (v APIEndpoint) IsValid() bool {
 
 // String returns a formatted version HOST:PORT of this APIEndpoint.
 func (v APIEndpoint) String() string {
-	return fmt.Sprintf("%s:%d", v.Host, v.Port)
+	return net.JoinHostPort(v.Host, fmt.Sprint(v.Port))
```

Comment:
I'm a bit torn, since fmt.Sprint(), does the right thing, since the default format for an int32 is `%d`, but I also think it's a good idea to be more explicit rather than implicit here to avoid any confusion.

Reply:
With `strconv.Itoa`, the port number would have to be first converted from `int32` to `int` and then to `string`. So I preferred `fmt.Sprint` since there is one less type conversion this way.
Totally agree with being more explicit by using `%d` here.

Reply:
Made the requested changes. Ready for review again.